mesonbuild/modules/gnome.py

```diff
@@ -157,8 +157,12 @@ def compile_resources(self, state, args, kwargs):
             output = args[0] + '.gresource'
             name = args[0] + '_gresource'
         else:
-            output = args[0] + '.c'
-            name = args[0] + '_c'
+            if 'cpp' in state.project_args.keys():
```

Comment:
I think this should rather check in `state.environment.coredata.compilers.keys()`. I'm not sure `state.project_args` contains anything unless the project does `add_project_arguments()`. Also I think it's preferable to still generate .c file in case both c and cpp compilers are available, I assume C is faster to compile, no?

Reply:
Checking `state.environment.coredata.compilers.keys()` won't work. `state.environment.coredata.compilers` contains several `OrderedDict` (two in my test). I don't know which one to choose.

Reply:
Ah right, it's per-machine. Try: `state.environment.coredata.compilers.host.keys()`

Reply:
Changed.
> Also I think it's preferable to still generate .c file in case both c and cpp compilers are available, I assume C is faster to compile, no?

Why? C is generally faster to compile, but for the resource file, the difference might not be noticeable. That requires testing. Also, generating the second resource source file might take longer than what you will gain on compile time. Then, there is also the issue with different flags for compilers...

Reply:
There are at least two reasons. First of all C is more tested so using it is the more reliable option.

Second of all if we do use the C++ compiler and plain C people notice that, they will complain and file bugs. As an example suppose you have a plain C project that uses a subproject that is implemented in C++. In this case the resource file would be compiled with a C++ compiler even though the user probably does not want that. This also changes the linker from C to C++, which is again something people might not want. IIRC there are platforms where having an unnecessary dependency on libstdc++ is a big no-no.

Reply:
> generating the second resource source

No, only generate one of course if you have a C compiler you generate .c and if not but have C++ then you generate .cpp instead.

> This also changes the linker from C to C++, which is again something people might not want. IIRC there are platforms where having an unnecessary dependency on libstdc++ is a big no-no.

Very true, it's indeed worse than I thought. For example gtk+ uses gnome.compile_resources() and has C++ compiler for a single unit test that checks that gtk headers works with C++ compiler, linking gtk on libstdc++ will definitely be rejected.